### PR TITLE
NIAD-1123 Docker Compose files

### PR DESCRIPTION
**Pull request by z-uddin**
* Added .dockerignore, docker-compose.yml, docker-compose-tests.yml and Dockerfile
* Renamed root project name to "gpc-consumer"
* Updated README to detail how to run local service and tests

### CODE REVIEW

File: docker/docker-compose-tests.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+version: '3.4'
+
+services:
+  gpc-consumer:
+    container_name: tests
+    build:
+      target: build
+    entrypoint: "gradle check"
+    environment:
+      - DISABLE_TEST_CONTAINERS=true
```
Comment (Broom94) [inline docker/docker-compose-tests.yml:11]: Do we have test containers?

File: service/build.gradle [hunk 2]
```diff
@@ -19,7 +18,12 @@ ext {
 dependencies {
 	implementation 'org.springframework.boot:spring-boot-starter-actuator'
 	implementation 'org.springframework.cloud:spring-cloud-starter-gateway'
+
+	// Test
 	testImplementation 'org.springframework.boot:spring-boot-starter-test'
+	testImplementation "org.assertj:assertj-core:3.18.1"
+	testImplementation "org.testcontainers:testcontainers:1.15.1"
+	testImplementation 'org.junit.jupiter:junit-jupiter-params:5.7.0'
```
Comment (Broom94) [inline service/build.gradle:26]: I don't think we need these additional imports

**Review by Broom94 — CHANGES_REQUESTED** [review]

File: service/settings.gradle [hunk 3]
```diff
@@ -1 +1 @@
-rootProject.name = 'consumer'
+rootProject.name = 'gpc-consumer'
```
Comment (Broom94) [inline service/settings.gradle:1]: nice spot

File: docker/docker-compose-tests.yml (hunk 1 above)
Comment (z-uddin) [reply docker/docker-compose-tests.yml:11]: Not at the moment, no good spot will remove this

**Comment by Broom94** [comment]
Can you add an env file for the start-script like gp2gp is done?

**Comment by Broom94** [comment]
Add .DS_Store to gitignore

File: docker/docker-compose-tests.yml [hunk 4]
```diff
@@ -6,6 +6,4 @@ services:
     container_name: tests
     build:
       target: build
```
Comment (Broom94) [inline docker/docker-compose-tests.yml:8]: What are we using this file for?

Comment (z-uddin) [reply docker/docker-compose-tests.yml:8]: Now removed as a part of this PR

Comment (z-uddin) [reply docker/docker-compose-tests.yml:8]: Believe this file will be used once the pipeline is built, to change the entry point to gradle check to run all tests.

**Review by Broom94 — APPROVED** [review]

**Review by msolecka — APPROVED** [review]

**Pull request merged**